### DISCUSSION TITLE
Integrating LangFuse

### DISCUSSION THREAD

**Discussion by algorithmx** [langgenius/dify#5083] [General]
I want to integrate [LangFuse](https://github.com/langfuse/langfuse) into dify source code. Where do you think is the best place to add the LangFuse's [`@observe()`](https://langfuse.com/blog/2024-04-python-decorator#goals) decorator?

**Comment by patryk20120** [comment]
What do you need LangFuse for?

**Reply by algorithmx** [reply]
I am planning to share my Dify workflows. So I need LangFuse for a detailed log and statistics of the services provided by Dify. Currently the monitoring and statistics functionalities in Dify is far from satisfactory.
